bigquery/google/cloud/bigquery/table.py

```diff
@@ -456,52 +458,77 @@ def table_type(self):
         """
         return self._properties.get('type')
 
+    @property
+    def time_partitioning(self):
+        """google.cloud.bigquery.table.TimePartitioning: Configures time-based
+        partitioning for a table.
+
+        Raises:
+            ValueError:
+                If the value is not :class:`TimePartitioning` or :data:`None`.
+        """
+        prop = self._properties.get('timePartitioning')
+        if prop is not None:
+            return TimePartitioning.from_api_repr(prop)
+
+    @time_partitioning.setter
+    def time_partitioning(self, value):
+        if value is not None and isinstance(value, TimePartitioning):
```

Comment:
Is checking `isinstance` necessary? We've been removing a lot of client-side validation like that.